woocommerce/woocommerce
# [ci-jobs] Add the optional property for jobs

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/47223.

Added support for optional jobs in the `ci-jobs` utility. The workflow part (`ci.yml`) will be added in a future PR, so the changes have no effect on the running checks yet.

Jobs can now be configured as optional by adding the `optional: true` property.
The default will be `false`, meaning the job will be required.
The performance tests jobs were configured as optional.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Unit tests should cover the changes and should of course pass.
2. Locally, run the ci-jobs util and check the output. The optional jobs have the `(optional)` added to their name:
    - `pnpm utils ci-jobs`
    - The 2 performance tests jobs are configured with `optional: true` and should appear as optional in the list
    - The 2 API tests jobs are configured with `optional: false`. They are mandatory.
    - The rest of the jobs have no `optional` configuration and should be treated as mandatory.
